### PR TITLE
Inheritance fixes

### PR DESCRIPTION
**Pull request by weisswurstkanone**
see #4997

### CODE REVIEW

File: models/DataObject/Concrete/Dao/InheritanceHelper.php [hunk 1]
```diff
@@ -197,51 +214,55 @@ public function addRelationToCheck($fieldname, $fieldDefinition, $queryfields =
     /**
      * @param $oo_id
      * @param bool $createMissingChildrenRows
+     * @params array $params
```
Comment (blankse) [inline models/DataObject/Concrete/Dao/InheritanceHelper.php]: Only `@param`

File: models/DataObject/Concrete/Dao/InheritanceHelper.php [hunk 2]
```diff
@@ -357,20 +389,46 @@ public function doDelete($objectId)
         }
     }
 
+
+    /**
+     * @param $result
+     * @param $language
+     * @param $column
```
Comment (blankse) [inline models/DataObject/Concrete/Dao/InheritanceHelper.php]:
```
     * @param array $result
     * @param string $language
     * @param string $column
```

Comment (weisswurstkanone) [reply models/DataObject/Concrete/Dao/InheritanceHelper.php]: thx!

File: models/DataObject/Concrete/Dao/InheritanceHelper.php [hunk 3]
```diff
@@ -32,6 +33,8 @@ class InheritanceHelper
 
     const ID_FIELD = 'oo_id';
 
+    const QUERY_ID_FIELD = 'ooo_id';
```
Comment (brusch) [inline models/DataObject/Concrete/Dao/InheritanceHelper.php]: Maybe we should name this constant `DEFAULT_QUERY_ID_COLUMN` which is more specific.

Comment (weisswurstkanone) [reply models/DataObject/Concrete/Dao/InheritanceHelper.php]: done

File: models/DataObject/Objectbrick/Data/Dao.php [hunk 4]
```diff
@@ -247,7 +247,10 @@ public function save(DataObject\Concrete $object, $params = [])
         $this->db->insertOrUpdate($querytable, $data);
 
         if ($inheritanceEnabled) {
-            $this->inheritanceHelper->doUpdate($object->getId(), true);
+            $this->inheritanceHelper->doUpdate($object->getId(), true,
+                ['relationContext' => [
```
Comment (brusch) [inline models/DataObject/Objectbrick/Data/Dao.php]: This should be `inheritanceRelationContext` right?

File: models/DataObject/Concrete/Dao/InheritanceHelper.php [hunk 5]
```diff
@@ -137,10 +147,16 @@ public function __construct($classId, $idField = null, $storetable = null, $quer
         } else {
             $this->idField = $idField;
         }
+
+        if ($queryIdField == null) {
```
Comment (brusch) [inline models/DataObject/Concrete/Dao/InheritanceHelper.php]: Maybe `===` here.

**Pull request merged**